scripts/docker-helper/cht-docker-compose.sh

```diff
@@ -0,0 +1,442 @@
+    http_code=$(curl -k --silent --show-error --head "$chtUrl" --write-out '%{http_code}' | tail -n1)
+    if [ "$http_code" != "200" ]; then
+      echo "CHT is returning $http_code instead of 200."
+    fi
+  else
+    echo "Port $chtPort is not open on $chtIp"
+  fi
+}
+
+validate_env_file(){
+  envFile=$1
+  if [ ! -f "$envFile" ] || [[ ! "$(file $envFile)" == *"ASCII text"* ]]; then
+    echo "File not found or not a text file: $envFile"
+  else
+    # TODO- maybe grep for env vars we're expecting? Blindly including
+    # is a bit promiscuous
+    # shellcheck disable=SC1090
+    . "${envFile}"
+    if [ -z "$COMPOSE_PROJECT_NAME" ] || [ -z "$CHT_HTTP" ] || [ -z "$CHT_HTTPS" ]; then
+      echo "Missing env value in file: COMPOSE_PROJECT_NAME, CHT_HTTP or CHT_HTTPS"
+    elif [[ "$COMPOSE_PROJECT_NAME" =~ [A-Z] ]];then
+      echo "COMPOSE_PROJECT_NAME can not have upper case: $COMPOSE_PROJECT_NAME"
+    fi
+  fi
+}
+
+get_running_container_count(){
+  result=0
+  containers="$1"
+  IFS=' ' read -ra containersArray <<< "$containers"
+  for container in "${containersArray[@]}"
+  do
+    if [ "$( docker ps -f name="${container}" | wc -l )" -eq 2 ]; then
+        (( result++ ))
+    fi
+  done
+  echo "$result"
+}
+
+get_global_running_container_count(){
+  # shellcheck disable=SC1083
+  if [ "$( docker ps --format={{.Names}} | wc -l )" -gt 0 ]; then
+    docker ps --format={{.Names}} | wc -l
+  else
+    echo "0"
+  fi
+}
+
+volume_exists(){
+  project=$1
+  volume="${project}_medic-data"
+  if [ "$( docker volume inspect "${volume}" 2>&1  | wc -l )" -eq 2 ]; then
+    echo "0"
+  else
+    echo "1"
+  fi
+}
+
+get_images_count(){
+  images="$1"
+  result=0
+  IFS=' ' read -ra imagesArray <<< "$images"
+  for image in "${imagesArray[@]}"
+  do
+    if [ "$( docker image ls --format {{.Repository}}:{{.Tag}} | grep -c "${image}" )" -eq 1 ]; then
+        (( result++ ))
+    fi
+  done
+  echo "$result"
+}
+
+pull_images(){
+  images="$1"
+  IFS=' ' read -ra imagesArray <<< "$images"
+  for image in "${imagesArray[@]}"
+  do
+    docker image pull "${image}" >/dev/null 2>&1
+  done
+}
+
+get_docker_compose_yml_path(){
+  if [ -f docker-compose-developer.yml ]; then
+    echo "docker-compose-developer.yml"
+  elif [ -f ../../docker-compose-developer.yml ]; then
+    echo "../../docker-compose-developer.yml"
+  else
+    return 0
+  fi
+}
+
+docker_up_or_restart(){
+  # some times this function called too quickly after a docker change, so
+  # we sleep 3 secs here to let the docker container/volume stabilize
+  sleep 3
+
+  envFile=$1
+  composeFile=$2
+
+  # haproxy never starts on first "up" call, so you know, call it twice ;)
+  docker-compose --env-file "${envFile}" -f "${composeFile}" down >/dev/null 2>&1
+  docker-compose --env-file "${envFile}" -f "${composeFile}" up -d >/dev/null 2>&1
+  docker-compose --env-file "${envFile}" -f "${composeFile}" up -d >/dev/null 2>&1
+}
+
+install_local_ip_cert(){
```

Comment:
Often people terminate with signed TLS in the reverse proxy and run self signed within local network without validation.

Either way, perhaps we should move these downloads into the entrypoint script.

Reply:
> Often people terminate with signed TLS in the reverse proxy and run self signed within local network without validation.

For local app dev hosting, the `medic-os` + `haproxy` stack is what we have to work with.  As such, following [our own instructions](https://docs.communityhealthtoolkit.org/apps/guides/hosting/ssl-cert-install/) to add a cert to v2 arch, overwriting the self signed cert in nginx is the way to do this for now.  

> Either way, perhaps we should move these downloads into the entrypoint script.

Given we can't revise the v2 docker images with a new entrypoint, are you suggesting we use the `command` feature with in the yaml for `docker-compose-development.yml`?  My concern about doing this was two fold: 

1. `medic-os` is especially complex (60+ processes when fully running) and I didn't want to break any existing entry points
2. it needs to be stable enough (eg not mid reboot when failing a health check) to download the local-ip.co certs.  

But I'm still pretty new to Docker - I'm open to ideas! You think given the arch v2  constraints (which I suspect I didn't fully convey - sorry!) we safely can do it as you suggest?